shared/js/intl_helper.js

```diff
@@ -0,0 +1,148 @@
+(function(global) {
+  'use strict';
+
+  /*************************
+
+  IntlHelper.defineObject('shortDate', 'datetime', {
+    year: 'numeric',
+    month: 'long',
+  });
+
+  function setValue() {
+    var formatter = IntlHelper.getObject('shortDate');
+    console.log(formatter.format(new Date()));
+  }
+
+  setValue();
+
+  IntlHelper.on('shortDate', setValue);
+  
+   ****************************/
+  var objectCache;
```

Comment:
I switched it to 'objects' instead of 'formatters' because Collator is not a formatter, and I want to also support other intl objects in the future (DisplayNames may be one) etc.